.github/workflows/on-release.yml

```diff
@@ -15,31 +15,31 @@ jobs:
     timeout-minutes: 10
 
     steps:
-    - name: Checkout
+    - name: "Checkout"
       uses: actions/checkout@v4
       with:
         fetch-depth: 0
 
-    - name: Set up Python 3.12
-      uses: actions/setup-python@v4
+    - name: "Set up Python"
+      uses: actions/setup-python@v5
       with:
-        python-version: '3.12'
+        python-version-file: "pyproject.toml"
 
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install poetry
+    - name: "Install uv"
+      uses: astral-sh/setup-uv@v6
+      with:
+        version: "0.7.21"
+        enable-cache: true
```

Comment:
Do we really need cache enabled here?

Reply:
yes, to speedup setup step between workflows and runs